docs/DeveloperGuide.md

```diff
@@ -2,36 +2,530 @@
+
+<img src="UML Diagrams/ListCommandDiagram.png" width="350">
+
+---
+
+## 3. Delete Command
+
+The "Delete" function is designed to enable users to remove specific drugs from the inventory based on the drug's name, 
+to remove drugs they no longer need, fully depleted or discontinued.
+
+**Design Considerations**
+
+1. **User-Specified Drug Name:** The function allows users to specify the drug name they want to delete from the
+   inventory.
+
+2. **Data Integrity:** It ensures that the deletion operation maintains the integrity of the inventory data structure,
+   updating it correctly.
+
+**Implementation and Rationale**
+
+This method is executed when the delete command is invoked. First, it attempts to delete a drug from the inventory 
+using the inventory.deleteDrug(this.keyToDelete) method. If successful, it retrieves the deleted entry.
+If the drug is successfully deleted, it returns a success message. 
+If the drug is not found (i.e., a DrugNotFoundException is thrown), it returns a failure message.
+
+---
+
+## 4. Help Command
+
+The command is responsible for showing users a list of all possible commands.
+
+**Design Considerations**
+
+The command was designed to print out and show how to use a list of all possible commands in a neat and concise way.
+
+**Implementation and Rationale**
+
+Command will use java's system out to print out all required information with a blank line in between for clarity.
+
+---
+
+## 5. Save Command
+
+The save command was made as a means to backup user entered drug data into the hard drive of the computer to ensure
+previously entered data is saved and accessable whenever the app is launched.
+
+### Design Considerations
+
+The save command had to be implemented in a way to enable direct writing of files onto the hard drive and a function had
+to be made to load said file back into the drug inventory upon starting the application.
+
+### Implementation
+
+There is a method to access the drugs within the inventory class. a separate method from the inventory class would then
+write the contents of these drugs back to the txt file for saving.
+
+Upon booting up the system, a method from the inventory class goes through the contents of the txt file and copies it to
+the inventory drug list.
+
+---
+
+## 6. addVendor Command
+
+The command was made to add vendors to a list of vendors so as to have access to it when needed.
+
+### Design Considerations
+
+The storage must be size adjustable based on the number of objects, in this case vendors placed into it.
+
+### Implementation
+
+In order to attain the adjustable storage based on numebr of objects, an ArrayList was used as additional vendors can be
+appended to the ArrayList whenever a new entry is required.
+
+---
+
+## 7. listVendors Command
+
+The command was made to list all vendors being tracked by the system in a neat way to the user
+
+### Design Considerations
+
+The possibility of an empty list had to be considered
```

Comment:
Would it be better to add more details on what is 'considered'? Like I guess you may want to handle excepeion here.